lib/run_loop/app.rb

```diff
@@ -69,20 +87,26 @@ def executable_name
 
     # Inspects the app's file for the server version
     def calabash_server_version
-      if valid?
-        path_to_bin = File.join(path, executable_name)
-        xcrun ||= RunLoop::Xcrun.new
-        hash = xcrun.exec(["strings", path_to_bin])
-        unless hash.nil?
-          version_str = hash[:out][/CALABASH VERSION: \d+\.\d+\.\d+/, 0]
-          unless version_str.nil? || version_str == ""
-            server_ver = version_str.split(":")[1].delete(' ')
-            RunLoop::Version.new(server_ver)
-          end
+      version = nil
+      executables.each do |executable|
+        version = strings(executable).server_version
+        break if version
+      end
+      version
```

Comment:
Wait, not completely wrong - `path/**/*` isn't completely recursive. Though I guess it doesn't really make sense to recurse as far down as app extensions since those _shouldn't_ really matter for validating whether calabash is linked.